cmd/nvidia-device-plugin/main.go

```diff
@@ -41,18 +41,24 @@ import (
 	"github.com/NVIDIA/k8s-device-plugin/internal/watch"
 )
 
+type options struct {
+	flags         []cli.Flag
+	configFile    string
+	kubeletSocket string
+}
+
 func main() {
-	var configFile string
+	o := &options{}
 
 	c := cli.NewApp()
 	c.Name = "NVIDIA Device Plugin"
 	c.Usage = "NVIDIA device plugin for Kubernetes"
 	c.Version = info.GetVersionString()
 	c.Action = func(ctx *cli.Context) error {
-		return start(ctx, c.Flags)
+		return start(ctx, o)
 	}
 
-	c.Flags = []cli.Flag{
+	o.flags = []cli.Flag{
```

Comment:
I would reverse this and set c.flags here and then update o.flags afterwards.

Reply:
I see, so flags defined before even defining c.App and then grouping all of the c.X assignments together. Yeah, that seems fine. The only thing we would "lose" is the niceness of having this near the top of the function:
```
	c.Name = "NVIDIA Device Plugin"
	c.Usage = "NVIDIA device plugin for Kubernetes"
```
but that is minor

Reply:
After looking at the changes required, I have decided to go with your suggestion.